lib/openhab/core/dsl/monkey_patch/types/decimal_type.rb

```diff
@@ -10,15 +10,51 @@ class Java::OrgOpenhabCoreLibraryTypes::DecimalType
   # rubocop:enable Style/ClassAndModuleChildren
 
   #
-  # Compare self to other using Java BigDecimal compare method
+  # Compare DecimalType to supplied object
   #
   # @param [Object] other object to compare to
   #
-  # @return [Boolean] True if have the same BigDecimal representation, false otherwise
+  # @return [Integer] -1,0,1 or nil depending on value supplied, nil comparison to supplied object is not possible.
   #
-  def ==(other)
-    return equals(other) unless other.is_a? Integer
+  def <=>(other)
+    logger.trace("#{self.class} #{self} <=> #{other} (#{other.class})")
+    case other
+    when Integer, Float
+      to_big_decimal.compare_to(Java::JavaMath::BigDecimal.new(other))
+    when Java::OrgOpenhabCoreTypes::UnDefType
+      1
+    else
+      other = other.state if other.respond_to? :state
+      compare_to(other)
+    end
+  end
+
+  #
+  # Coerce objects into a DecimalType
+  #
+  # @param [Object] other object to coerce to a DecimalType if possible
+  #
+  # @return [Object] Numeric when applicable
+  #
+  def coerce(other)
+    logger.trace("Coercing #{self} as a request from  #{other.class}")
+    case other
+    when Integer, Float
```

Comment:
Same question on Numeric vs the concrete types.

Reply:
see above.